### PR TITLE
Test re-deploy if any deploy scripts change

### PR DESCRIPTION
**Pull request by dfarrell07**
Test re-deploying on an already-deployed environment when any deployment
script is modified, not just e2e.sh.

Closes: #392

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

### CODE REVIEW

**Comment by dfarrell07** [comment]
> I'm not sure, we should only recheck if a 'sh' file changed and only one that's pertinent to the job at hand.. If a YAML file changed, rerun would just waste time but not really check anything since the logic is not there.

I was following your suggestion with this logic ;)

> A simple fix could be to trigger on "${CHANGED_FILES_PR[@]}" =~ "scripts/kind-e2e/"

https://github.com/submariner-io/submariner/issues/392#issuecomment-598070298

**Comment by dfarrell07** [comment]
> > I'm not sure, we should only recheck if a 'sh' file changed and only one that's pertinent to the job at hand.. If a YAML file changed, rerun would just waste time but not really check anything since the logic is not there.
> 
> I was following your suggestion with this logic ;)
> 
> > A simple fix could be to trigger on "${CHANGED_FILES_PR[@]}" =~ "scripts/kind-e2e/"
> 
> [#392 (comment)](https://github.com/submariner-io/submariner/issues/392#issuecomment-598070298)

But you're right, I push a more selective version and it looks much better to me. Thanks for the useful review!

File: .travis.yml [hunk 1]
```diff
@@ -26,8 +26,10 @@ script:
   - set -o pipefail;
     $CMD 2>&1 | ts '[%H:%M:%.S]' -s
 after_success:
-  - if [[ "${CHANGED_FILES_PR[@]}" =~ "scripts/kind-e2e/e2e.sh" ]]; then
-      echo "scripts/kind-e2e/e2e.sh was modified, testing recurring run on already deployed infrastructure.";
+  - if [[ ( $CMD =~ "e2e" && "${CHANGED_FILES_PR[@]}" =~ "scripts/kind-e2e/e2e.sh" ) ||
+          ( $CMD =~ "helm" && "${CHANGED_FILES_PR[@]}" =~ "scripts/kind-e2e/lib_helm_deploy_subm.sh" ) ||
+          ( ! $CMD =~ "helm" && "${CHANGED_FILES_PR[@]}" =~ "scripts/kind-e2e/lib_operator_deploy_subm.sh" ) ]]; then;
+      echo "scripts/kind-e2e/*.sh files used by this job were modified, testing recurring run on already deployed infrastructure.";
```
Comment (mkolesnik) [inline .travis.yml]:
```suggestion
  - if [[ $CMD =~ "e2e" &&
          ( "${CHANGED_FILES_PR[@]}" =~ "scripts/kind-e2e/e2e.sh" ) ||
          ( $CMD =~ "helm" && "${CHANGED_FILES_PR[@]}" =~ "scripts/kind-e2e/lib_helm_deploy_subm.sh" ) ||
          ( ! $CMD =~ "helm" && "${CHANGED_FILES_PR[@]}" =~ "scripts/kind-e2e/lib_operator_deploy_subm.sh" ) ]]; then;
      echo "scripts/kind-e2e/*.sh files used by this job were modified, testing recurring run on already deployed infrastructure.";
```

File: .travis.yml [hunk 2]
```diff
@@ -26,8 +26,11 @@ script:
   - set -o pipefail;
     $CMD 2>&1 | ts '[%H:%M:%.S]' -s
 after_success:
-  - if [[ "${CHANGED_FILES_PR[@]}" =~ "scripts/kind-e2e/e2e.sh" ]]; then
-      echo "scripts/kind-e2e/e2e.sh was modified, testing recurring run on already deployed infrastructure.";
+  - if [[ $CMD =~ "e2e" &&
+          ( "${CHANGED_FILES_PR[@]}" =~ "scripts/kind-e2e/e2e.sh" ) ||
+          ( $CMD =~ "helm" && "${CHANGED_FILES_PR[@]}" =~ "scripts/kind-e2e/lib_helm_deploy_subm.sh" ) ||
+          ( ! $CMD =~ "helm" && "${CHANGED_FILES_PR[@]}" =~ "scripts/kind-e2e/lib_operator_deploy_subm.sh" ) ]]; then
```
Comment (mkolesnik) [inline .travis.yml]:
```suggestion
  - if [[ $CMD =~ "e2e" &&
          (
            ("${CHANGED_FILES_PR[@]}" =~ "scripts/kind-e2e/e2e.sh") ||
            ($CMD =~ "helm" && "${CHANGED_FILES_PR[@]}" =~ "scripts/kind-e2e/lib_helm_deploy_subm.sh") ||
            (! $CMD =~ "helm" && "${CHANGED_FILES_PR[@]}" =~ "scripts/kind-e2e/lib_operator_deploy_subm.sh")
          ) ]]; then
```

**Pull request merged**